ebib-utils.el

```diff
@@ -392,22 +392,33 @@ transformation function returns something that can be displayed."
   :type '(repeat (cons (string :tag "Field")
                        (function :tag "Transform function"))))
 
-(defcustom ebib-TeX-markup-replace-alist '(("{" . "")
-                                           ("}" . "")
-                                           ("``" . "“")
+(defcustom ebib-TeX-markup-replace-alist '(("``" . "“")
 					   ("`" . "‘")
 					   ("''" . "”")
-					   ("'" . "’"))
+					   ("'" . "’")
+					   ("\\\\textit\\(?:\\[.*\\]\\)?{\\(.*?\\)}"
+					    . ebib--convert-tex-italics)
+					   ("\\\\emph\\(?:\\[.*\\]\\)?{\\(.*?\\)}"
+					    . ebib--convert-tex-italics)
+					   ("\\\\textbf\\(?:\\[.*\\]\\)?{\\(.*?\\)}"
+					    . ebib--convert-tex-bold)
+					   ("\\\\textsc\\(?:\\[.*\\]\\)?{\\(.*?\\)}"
+					    . ebib--convert-tex-small-caps)
+					   ("\\\\[[:alpha:]]+\\(?:\\[.*\\]\\)?{\\(.*?\\)}" . "\\1")
```

Comment:
I used `[a-zA-Z*]` as the character class because (La)TeX commands can only consist of ASCII letters and may optionally end in an asterisk. (Actually, technically I believe the asterisk is some kind of argument, but I don't understand the particulars). `[:alpha:]` misses the asterisk and includes accented characters and characters from non-Latin alphabets, which AFAIK are not allowed in command names.

The group capturing optional arguments should also be modified a bit, because commands can have more than one optional argument. The original regex has `\\(?:\\[.*?\\]\\)*`, which I think works better.

Or actually, come to think of it, your version should also work, because `\\[.*\\]` will match the *longest* sequence of characters starting with `[` and ending with `]`, which should capture all optional arguments in one go. It might even be faster. Ok, in that case, leave it as it is and consider it an improvement. :wink:

Boy, I wish I had invested time in writing some tests for these kinds of things... :smile:

Reply:
Sorted. Thanks for explaining!